components/homme/src/share/cxx/EulerStepFunctorImpl.hpp

```diff
@@ -29,11 +29,10 @@ namespace Kokkos {
 struct Real2 {
   Homme::Real v[2];
   KOKKOS_FORCEINLINE_FUNCTION Real2 () { v[0] = v[1] = 0; }
-  KOKKOS_FORCEINLINE_FUNCTION Real2 operator+= (const Real2& o) const {
```

Comment:
Why is this a bug? Say, it was called with Real2 aa+=bb; then first version overwrites aa, second version just changes values in aa, but why is this a bug?

Reply:
The old version was not modifying the object it was called on. It was returning a+b, but a was unchanged.